### ISSUE TITLE
Idea: PublicBuilder presets for common usecases

### ISSUE THREAD

**Issue by Firstyear** [parallaxsecond/rust-tss-esapi#499]
There are some common cases we could add to public builder to save users from having to determine all the required feature flags and combinations needed. An example could be PublicBuilder::new_sealed_data_object() where we have this yield and create a value objectattributes and public that meet the requirements. Is that something that would be wanted? I'm sure there are other use cases too.

**Comment by wiktor-k** [comment]
I think it's a good idea. AFAIUI it would pre-populate the builder with some known good combinations and still require specifying e.g. exact algorithms...?

**Comment by Firstyear** [comment]
Yes that's what I was thinking. Some examples could be from the tpm spec section 25.1.5, such as "sealed data object", "general purpose key", "storage key".

**Comment by Superhepper** [comment]
Yeah, that can be useful as long as we do not specify key sizes or algorithms
